app/src/main/java/com/lagradost/cloudstream3/ui/download/DownloadFragment.kt

```diff
@@ -236,7 +237,9 @@ class DownloadFragment : Fragment() {
             }
 
             DOWNLOAD_ACTION_LOAD_RESULT -> {
-                activity?.loadResult(click.data.url, click.data.apiName)
+                if (context?.isNetworkAvailable() == true) {
```

Comment:
I dont understand this, why? It seams more like a anti-feature than a feature, and makes the experience inconsistent.  

Especially considering 
1. Context may be null
2. isNetworkAvailable may fail (due to security or permission) 
3. Weird network setups creating a false negative
4. The result may even be cached so no network is needed

Reply:
> I dont understand this, why? It seams more like a anti-feature than a feature, and makes the experience inconsistent.
> 
> Especially considering
> 
> 1. Context may be null
> 2. isNetworkAvailable may fail (due to security or permission)
> 3. Weird network setups creating a false negative
> 4. The result may even be cached so no network is needed

@fire-light42 

That is fair I suppose my reasoning for it is when I did not have any network, I kept accidently clicking on the thumbnail. which just took me to an api failed page which was quite annoying but I can revert if you want. But to address your concerns:

1. that was unintentional, I had it originally as if if (context?.isNetworkAvailable() != false) which would cover if it is null (I think) not sure why I changed it to == true lol
2. we can address this so it still works if it fails like that
3. I'm not sure why this would happen? From my understanding network checks like this are fairly common, including in many other applications, at least from what I found
4. that one is a good point I guess, but when it isn't it can still get quite annoying to fail with the network when we could just stop it loading in the first place. My other idea was if no network, to instead of navigating just show a toast or snackbar that it has no network detected instead. Nonetheless if you want me to I will revert.

Reply:
I consider consistency to be better, and I dont trust any network connection status to be valid or invalid before actually doing the get request. 

If you feel like this is wrong, then revert this and create a separate PR for this to extend or modify the functionally. I would at least require a toast to be shown if you really want this feature, as otherwise it will confuse users.

Reply:
Done